inst/include/dust/tools.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef DUST_TOOLS_HPP
+#define DUST_TOOLS_HPP
+
+#include <algorithm>
+#include <numeric>
+
+template <typename real_t>
+void resample_weight(typename std::vector<real_t>::const_iterator w,
+                     size_t n, real_t u, size_t offset,
+                     typename std::vector<size_t>::iterator idx) {
```

Comment:
Correct for these iterators to be copies?

Reply:
yes, this is normal style I think for iterators (the iterator itself will just be a `real_t *` by the time it gets here)